jena-fuseki2/jena-fuseki-core/src/main/java/org/apache/jena/fuseki/servlets/prefixes/ActionProcPrefixes.java

```diff
@@ -0,0 +1,228 @@
+                    String namespace = x.orElse(NO_PREFIX_NS);
+
+                    JsonObject jsonObject = new JsonObject(prefix, namespace);
+
+                    // Build the response.
+                    action.setResponseContentType(WebContent.contentTypeJSON);
+                    action.getResponseOutputStream().print(namespace);
+                    // Indicate success
+                    FmtLog.info(action.log, "[%d] %s -> %s", action.id, prefix, jsonObject.toString());
+                    transactional.commit();
+                    ServletOps.success(action);
+                    return;
+                }
+            }
+            if (prefix == null && uri != null) {
+                if (uri.isEmpty()) {
+                    ServletOps.errorBadRequest("Empty URI!");
+                    return;
+                }
+                else if (!PrefixUtils.uriIsValid(uri)) {
+                    ServletOps.errorBadRequest("URI contains illegal characters!");
+                    return;
+                }
+                else {
+                    //fetchPrefix
+                    List<String> prefixList =storage.fetchPrefix(uri);
+                    JsonArray prefixJsonArray = new JsonArray();
+                    for (String p : prefixList) {
+                        com.google.gson.JsonObject jsonObject2 = new com.google.gson.JsonObject();
+                        jsonObject2.addProperty("prefix", p);
+                        jsonObject2.addProperty("uri", uri);
+                        prefixJsonArray.add(jsonObject2);
+                        FmtLog.info(action.log, "[%d] - %s", action.id, new JsonObject(p, uri));
+                    }
+                    // Build the response.
+                    action.setResponseContentType(WebContent.contentTypeJSON);
+                    action.getResponseOutputStream().print(String.valueOf(prefixJsonArray));
+                    // Indicate success
+                    transactional.commit();
+                    ServletOps.success(action);
+                    transactional.end();
+                    return;
+                }
+            }
+
+        } catch (RuntimeException | IOException ex) {
+            try { transactional.abort(); }
+            catch (Throwable th ) {
+                FmtLog.warn(action.log, th, "[%d] GET prefix = %s", action.id);
+            }
+            ServletOps.errorOccurred(ex);
+        } finally {
+            transactional.end();
+        }
+    }
+
+    @Override
+    public void doPost (HttpAction action) {
+        validatePost(action);
+        Transactional transactional = storage.transactional();
+
+        transactional.begin(TxnType.WRITE);
+        try {
+            String prefix = action.getRequestParameter("prefix");
+            String uri = action.getRequestParameter("uri");
+
+            if(prefix.isEmpty()) {
+                String prefixToRemove = action.getRequestParameter("removeprefix");
```

Comment:
Again same comment about overloading `POST` to be both update and delete while also having a separate `DELETE` operation, would be cleaner to keep `POST` as update only